### DISCUSSION TITLE
How to change logout route, different for each panel.

### DISCUSSION THREAD

**Discussion by piyushdongre-dot-com** [filamentphp/filament#9989] [Help] [answered]
Hi,
I wan to change logout route for each panel to something custom. Currently it takes to <panel-nam>/login.

Thanks

**Comment by leandrocfe** [comment]
You can use this
```php
$panel
->userMenuItems([
    'logout' => MenuItem::make()->url(function () {

        //get current panel, if you need
        $panel = filament()->getCurrentPanel();

        return 'custom-logout-url..';
    }),
    // ...
])
```

**Reply by piyushdongre-dot-com** [reply]
After changing the route like `return '/';  it is making a POST request to the route. 

I am getting below error.
The POST method is not supported for route /. Supported methods: GET, HEAD.

Am I required to make these changes in web.php

**Reply by leandrocfe** [reply]
Yes, you can create custom routes in the web.php

**Reply by piyushdongre-dot-com** [reply]
Also, by default it doesnt logout the User. 

I had to add `Auth::logout();`

**Reply by leandrocfe** [reply, marked as answer]
```php
use Filament\Facades\Filament;
Filament::auth()->logout();

session()->invalidate();
session()->regenerateToken();
```
?
